sdk/test/trace/sampler_benchmark.cc

```diff
@@ -0,0 +1,211 @@
+	{
+		benchmark::DoNotOptimize(AlwaysOnSampler());
+	}
+}
+BENCHMARK(BM_AlwaysOnSamplerConstruction);
+
+void BM_ParentOrElseSamplerConstruction(benchmark::State &state)
+{
+	while (state.KeepRunning())
+	{
+		benchmark::DoNotOptimize(ParentOrElseSampler(std::make_shared<AlwaysOnSampler>()));
+	}
+}
+BENCHMARK(BM_ParentOrElseSamplerConstruction);
+
+void BM_ProbabilitySamplerConstruction(benchmark::State &state)
+{
+	while (state.KeepRunning())
+	{
+		benchmark::DoNotOptimize(ProbabilitySampler(0.01));
+	}
+}
+BENCHMARK(BM_ProbabilitySamplerConstruction);
+
+// Sampler used as a baseline to compare with other samplers
+void BM_AlwaysOffSamplerShouldSample(benchmark::State &state)
+{
+	AlwaysOffSampler sampler;
+
+	BenchmarkShouldSampler(sampler, state);
+}
+BENCHMARK(BM_AlwaysOffSamplerShouldSample);
+
+// Sampler used as a baseline to compare with other samplers
+void BM_AlwaysOnSamplerShouldSample(benchmark::State &state)
+{
+	AlwaysOnSampler sampler;
+
+	BenchmarkShouldSampler(sampler, state);
+}
+BENCHMARK(BM_AlwaysOnSamplerShouldSample);
+
+void BM_ParentOrElseSamplerShouldSample(benchmark::State &state)
+{
+	ParentOrElseSampler sampler(std::make_shared<AlwaysOnSampler>());
+
+	BenchmarkShouldSampler(sampler, state);
+}
+BENCHMARK(BM_ParentOrElseSamplerShouldSample);
+
+void BM_ProbabilitySamplerShouldSample(benchmark::State &state)
+{
+	ProbabilitySampler sampler(0.01);
+
+	BenchmarkShouldSampler(sampler, state);
+}
+BENCHMARK(BM_ProbabilitySamplerShouldSample);
+
+// Test to measure performance for span creation
+void BM_SpanCreation(benchmark::State &state)
+{
+	std::shared_ptr<std::vector<std::unique_ptr<SpanData>>> spans_received_off(
+		new std::vector<std::unique_ptr<SpanData>>);
+
+	std::unique_ptr<SpanExporter> exporter(new MockSpanExporter(spans_received_off));
+	auto processor = std::make_shared<SimpleSpanProcessor>(std::move(exporter));
+	auto tracer_off = std::shared_ptr<opentelemetry::trace::Tracer>(new Tracer(processor, std::make_shared<AlwaysOnSampler>()));
+
+	while (state.KeepRunning())
+	{
+		auto span_off_1 = tracer_off->StartSpan("span with AlwaysOn sampler");
+
+		span_off_1->SetAttribute("attr1", 3.1);
+
+		span_off_1->End();
+	}
+}
+BENCHMARK(BM_SpanCreation);
+
+// Test to measure performance overhead for no-op span creation
+void BM_NoopSpanCreation(benchmark::State &state)
+{
+	std::shared_ptr<std::vector<std::unique_ptr<SpanData>>> spans_received_off(
+		new std::vector<std::unique_ptr<SpanData>>);
+
+	std::unique_ptr<SpanExporter> exporter(new MockSpanExporter(spans_received_off));
+	auto processor = std::make_shared<SimpleSpanProcessor>(std::move(exporter));
+	auto tracer_off = std::shared_ptr<opentelemetry::trace::Tracer>(new Tracer(processor, std::make_shared<AlwaysOffSampler>()));
+
+	while (state.KeepRunning())
+	{
+		auto span_off_1 = tracer_off->StartSpan("span with AlwaysOff sampler");
+
+		span_off_1->SetAttribute("attr1", 3.1);  // Not recorded.
+
+		span_off_1->End();
+	}
+}
```

Comment:
Again there's a lot of similarities between BM_NoopSpanCreation and BM_SpanCreation. Can you extract any parts into a helper?

Reply:
Yep, done